.github/workflows/set-metadata.yml

```diff
@@ -0,0 +1,354 @@
+          fi
+          if ! [[ -n "${{ needs.get-howso-validator-enterprise-py-details.outputs.run-id }}" ]]; then
+            # Append run-id, run-type, build-date, build-title, head-sha, url to metadata JSON
+            details=$(echo "$details" | jq '. + {"howso-validator-enterprise-py": "{}"}')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".run_id="${{ needs.get-howso-validator-enterprise-py-details.outputs.run-id }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".run_type="${{ needs.get-howso-validator-enterprise-py-details.outputs.run-type }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".build_date="${{ needs.get-howso-validator-enterprise-py-details.outputs.build-date }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".build_title="${{ needs.get-howso-validator-enterprise-py-details.outputs.build-title }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".head_sha="${{ needs.get-howso-validator-enterprise-py-details.outputs.head-sha }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".url="${{ needs.get-howso-validator-enterprise-py-details.outputs.url }}"')
+          fi
+          if ! [[ -n "${{ needs.get-howso-synthesizer-py-details.outputs.run-id }}" ]]; then
+            # Append run-id, run-type, build-date, build-title, head-sha, url to metadata JSON
+            details=$(echo "$details" | jq '. + {"howso-engine-py": "{}"}')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".run_id="${{ needs.get-howso-synthesizer-py-details.outputs.run-id }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".run_type="${{ needs.get-howso-synthesizer-py-details.outputs.run-type }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".build_date="${{ needs.get-howso-synthesizer-py-details.outputs.build-date }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".build_title="${{ needs.get-howso-synthesizer-py-details.outputs.build-title }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".head_sha="${{ needs.get-howso-synthesizer-py-details.outputs.head-sha }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".url="${{ needs.get-howso-synthesizer-py-details.outputs.url }}"')
+          fi
+
+          echo "upstream-details=$(echo $details)" >> $GITHUB_OUTPUT
+  
+  set-version:
+    needs: ['set-metadata']
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.choose-version.outputs.version }}
+    steps:
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get previous git tag
+        id: previous-tag
+        run: |
+          tag=$(git for-each-ref --sort=-creatordate --count 5 --format="%(refname:short)" "refs/tags/" | grep -E "^[0-9]+\.[0-9]+\.[0-9]+" | head -n 1)
```

Comment:
Does this get both annotated and non-annotated tags? Want to make sure we don't miss any in this query.

Reply:
Also, creator date might not be the right sort to use here. Although with our current way of tagging releases, this shouldn't be an issue so probably not worth worrying about right now.

Reply:
This is actually pulled directly from the 3rd party action we were using: https://github.com/WyriHaximus/github-action-get-previous-tag/blob/master/main.js